cmd/tendermint/commands/init.go

```diff
@@ -18,28 +16,29 @@ var InitFilesCmd = &cobra.Command{
 
 func initFiles(cmd *cobra.Command, args []string) {
 	privValFile := config.PrivValidatorFile()
-	if _, err := os.Stat(privValFile); os.IsNotExist(err) {
-		privValidator := types.GenPrivValidatorFS(privValFile)
+	privValidator := types.GenPrivValidatorFS(privValFile)
+	if cmn.FileExists(privValFile) {
+		logger.Info("Already initialized", "priv_validator", config.PrivValidatorFile())
+	} else {
 		privValidator.Save()
+		logger.Info("Initialized tendermint", "privValidator")
+	}
 
-		genFile := config.GenesisFile()
-
-		if _, err := os.Stat(genFile); os.IsNotExist(err) {
-			genDoc := types.GenesisDoc{
-				ChainID: cmn.Fmt("test-chain-%v", cmn.RandStr(6)),
-			}
-			genDoc.Validators = []types.GenesisValidator{{
-				PubKey: privValidator.GetPubKey(),
-				Power:  10,
-			}}
-
-			if err := genDoc.SaveAs(genFile); err != nil {
-				panic(err)
-			}
+	genFile := config.GenesisFile()
+	if cmn.FileExists(genFile) {
+		logger.Info("Already initialized", "geneisis", genFile)
+	} else {
+		genDoc := types.GenesisDoc{
+			ChainID: cmn.Fmt("test-chain-%v", cmn.RandStr(6)),
 		}
+		genDoc.Validators = []types.GenesisValidator{{
+			PubKey: privValidator.GetPubKey(),
```

Comment:
shouldn't we load PrivVal if it exists?
otherwise, genesis file will have "incorrect" priv validator

Reply:
there was a TODO somewhere about it..

Reply:
I think if we used types.LoadOrGenPrivValidatorFS then we'd be set, and the function to use would be
https://github.com/tendermint/tendermint/blob/652d1e3de8a3f216299ca66c595cc66504930d28/types/priv_validator.go#L131-L140

Reply:
@melekes please take a look at the code above where we load it if it exists lest we generate it. Thank you for that suggestion!